### PR TITLE
make everything themeable

### PR DESCRIPTION
**Pull request by anthraxx**
This makes it posssible to theme everything logically grouped by
message types. This will also make it easier for future features
to keep a consistent way of coloring plus make every non-specific
coloring themeable automatically.

Direct explicit usage of colors should be avoided in future commits.

### CODE REVIEW

File: pwndbg/commands/got.py [hunk 1]
```diff
@@ -30,14 +28,19 @@ def got(name_filter=''):
...
-        print(red("NO JUMP_SLOT entries available in the GOT"))
+        print(message.error("NO JUMP_SLOT entries available in the GOT"))
         return
+
     if "PIE enabled" in pie_status:
         bin_text_base = pwndbg.memory.page_align(pwndbg.elf.entry())
 
-    print("\nGOT protection: %s | GOT functions: %d\n " % (green(relro_status), len(jmpslots)))
+    relro_color = message.off
+    if 'Partial' in relro_status:
```
Comment (disconnect3d) [inline pwndbg/commands/got.py:39]: Hope there isn't any edge case ;f

File: pwndbg/commands/heap.py [hunk 2]
```diff
@@ -124,10 +122,10 @@ def arenas():
         
         h = heap.get_region(addr)
         if not h:
-            print(red('Could not find the heap'))
+            print(message.error('Could not find the heap'))
             return
 
-        hdr = bold(fmt%(hex(addr) if addr else 'main'))
+        hdr = message.hint(fmt%(hex(addr) if addr else 'main'))
```
Comment (disconnect3d) [inline pwndbg/commands/heap.py]: space pls `fmt % `

**Comment by disconnect3d** [comment]
We should use:
```python
from pwndbg.color import message
```

instead of:
```python
import pwndbg.color.message as message
```

**Comment by anthraxx** [comment]
i agree, i was just mimic-ing what we did so far :stuck_out_tongue_closed_eyes:

File: pwndbg/commands/got.py (hunk 1 above)
Comment (anthraxx) [reply pwndbg/commands/got.py:39]: well thats how we stricty use  our wrapper/checksec.py so it must be the same output :yum:

File: pwndbg/commands/heap.py (hunk 2 above)
Comment (anthraxx) [reply pwndbg/commands/heap.py]: fixed

**Comment by disconnect3d** [comment]
This is awesome :heart:.

One more thing that comes to my mind: maybe we should change:
```python
print(message.notice(...))
```
Into:
```python
log.notice(...)
```

This could be then somehow connected to https://github.com/pwndbg/pwndbg/issues/245

**Comment by anthraxx** [comment]
yes we should, but lets do that in a separate PR attached to logging an not themeing. I volunteer to do it :smile:

**Comment by disconnect3d** [comment]
One more thing - fire `isort --recursive pwndbg` ;D

**Comment by anthraxx** [comment]
all implemented and isorted :smile_cat:

**Review by disconnect3d — APPROVED** [review]
lgtm :100:

**Comment by disconnect3d** [comment]
@anthraxx will merge just after check passes

**Comment by disconnect3d** [comment]
@anthraxx btw I haven't checked that and now I see that sth is not working fine; can u look for it?

```
[dc@dc:pwndbg|dev $%]$ gdb /bin/ls
pwndbg: loaded 158 commands. Type pwndbg [filter] for a list.
pwndbg: created $rebase, $ida gdb functions (can be used with print/break)
Reading symbols from /bin/ls...(no debugging symbols found)...done.
pwndbg> set exception-verbose on
Set whether to print a full stacktracefor exceptions raised in Pwndbg commands to True
pwndbg> entry
Temporary breakpoint 1 at 0x555555559060

Temporary breakpoint 1, 0x0000555555559060 in ?? ()
Traceback (most recent call last):
  File "pwndbg/pwndbg/events.py", line 119, in caller
    func()
  File "pwndbg/pwndbg/heap/__init__.py", line 18, in update
    import pwndbg.heap.ptmalloc
  File "pwndbg/pwndbg/heap/ptmalloc.py", line 14, in <module>
    from pwndbg.color.message import message
ImportError: cannot import name 'message'

If that is an issue, you can report it on https://github.com/pwndbg/pwndbg/issues
(Please don't forget to search if it hasn't been reported before)
PS: Pull requests are welcome
Traceback (most recent call last):
  File "pwndbg/pwndbg/prompt.py", line 33, in prompt_hook
    pwndbg.events.after_reload(start=False)
  File "pwndbg/pwndbg/events.py", line 163, in after_reload
    f()
  File "pwndbg/pwndbg/events.py", line 123, in caller
    raise e
  File "pwndbg/pwndbg/events.py", line 119, in caller
    func()
  File "pwndbg/pwndbg/heap/__init__.py", line 18, in update
    import pwndbg.heap.ptmalloc
  File "pwndbg/pwndbg/heap/ptmalloc.py", line 14, in <module>
    from pwndbg.color.message import message
ImportError: cannot import name 'message'
```

**Comment by zachriggle** [comment]
Can we add caching to these? I'm not sure what's the performance hit of going to GDB settings and pulling out config values for *every* color, *every* time we print it.

**Comment by anthraxx** [comment]
I can do some micro benchmark but it performs quite fast when stepping through nothing noticeable and I doubt it's from any interest compared to. All the things we do. Like dereference and stuff

**Comment by zachriggle** [comment]
Sure, I guess the parameters might not actually go down into the GDB native code and back out to Python.

**Pull request merged**